pkg/ccl/backupccl/backup_test.go

```diff
@@ -10017,3 +10017,119 @@ func TestBackupLatestInBaseDirectory(t *testing.T) {
 	query = fmt.Sprintf("BACKUP INTO LATEST IN %s", userfile)
 	sqlDB.Exec(t, query)
 }
+
+// TestBackupRandomizedMixVersion tests a sequence of sql statements on an
+// upgrading cluster. The timeline of the upgrading cluster will be randomly
+// chosen, implying a given sql statement may be executed in the old, mixed, or
```

Comment:
Why not just test every permutation every time? As-is, a dev might be more likely to be consider a failure here as "flaky" than "significant."